components/SearchContainer.js

```diff
@@ -2,56 +2,70 @@ import styled from "styled-components";
 import TextEntry from "./TextEntry";
 import Button from "./Button";
 import { useRouter } from "next/router";
+import { useStore } from "../hooks/useStore";
 
 export default function SearchContainer() {
   const router = useRouter();
+  const setArticles = useStore((state) => state.setArticles);
 
   async function handleSubmit(event) {
     event.preventDefault();
     const formdata = new FormData(event.target);
     const data = Object.fromEntries(formdata);
-    const { thema, source, publishedAt, country } = data;
-    const response = await fetch(
-      `/api/search?thema=${thema}&source=${source}&when=${publishedAt}&country=${country}`
-    );
+    const response = await fetch(`/api/search`, {
+      method: "POST",
+      body: JSON.stringify(data),
+    });
     const results = await response.json();
-    localStorage.setItem("articles", JSON.stringify(results));
+    const articles = results.map((result) => {
+      return { ...result, id: Math.random().toString(36).substring(2) };
+    });
+
+    setArticles(articles);
+
     router.push("/");
   }
+
   return (
     <StyledSearchContainer onSubmit={handleSubmit}>
       <TextEntry
-        label="thema"
-        id="thema"
-        name="thema"
-        placeholder="eg. politics"
+        label="keywords"
+        id="keywords"
+        name="keywords"
+        placeholder="eg. election, bitcoin"
       />
       <TextEntry
-        label="source"
-        id="source"
-        name="source"
-        placeholder="eg. nytimes"
+        label="sources"
+        id="sources"
+        name="sources"
+        placeholder="eg. nytimes, rolling stone"
       />
       <TextEntry
-        label="published at"
-        id="publishedAt"
-        name="publishedAt"
+        label="domains"
+        id="domains"
+        name="domains"
+        placeholder="eg. bbc.uk, politico.com"
+      />
+      <TextEntry
+        label="from"
+        id="from"
+        name="from"
         placeholder="eg. 2022-10-01"
       />
+      <TextEntry label="to" id="to" name="to" placeholder="eg. 2022-10-02" />
       <TextEntry
-        label="country"
-        id="country"
-        name="country"
-        placeholder="eg. england"
+        label="language"
+        id="language"
+        name="language"
+        placeholder="eg. de, en, es"
       />
+
```

Comment:
can you find a way where you don't repeat yourself this way? Try write more DRY code!

Reply:
yeah i will change it ;). Thanks!